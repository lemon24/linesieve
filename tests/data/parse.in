--section '\S+:$'
parse -s ^dict:$ -e '(?P<one>\d)(?P<two>\w)?'
parse -s ^list:$ -e '(\d)(\w)?'
parse -s ^dict-and-list:$ -e '(?P<one>\d)(\w)'
parse -s ^no-groups:$ -e '\d\w?'
parse -s ^json-dict:$ --json -e '(?P<one>\d)(?P<two>\w)?'
parse -s ^json-list:$ --json -e '(\d)(\w)?'
parse -s ^json-no-groups:$ --json -e '\d\w?'
parse -s ^patterns:$ -e '(?P<digit>\d)' -e '(?P<letter>[a-z])'
parse -s ^only-matching:$ -e '(\d)(\w)' -o
---
dict:
.1a.
.1.
no match
list:
.1a.
.1.
no match
dict-and-list:
.1a.
no match
no-groups:
.1a.
.1.
no match
json-dict:
.1a.
.1.
no match
json-list:
.1a.
.1.
no match
json-no-groups:
.1a.
.1.
no match
patterns:
.1.
.a.
...
only-matching:
1a
aa
