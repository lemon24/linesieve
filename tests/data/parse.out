dict:
one	1	two	a
one	1	two	
no match
list:
1	a
1	
no match
dict-and-list:
one	1
no match
no-groups:
1a
1
no match
json-dict:
{"one": "1", "two": "a"}
{"one": "1", "two": null}
no match
json-list:
["1", "a"]
["1", null]
no match
json-no-groups:
["1a"]
["1"]
no match
patterns:
digit	1
letter	a
...
only-matching:
1	a
